vyper/venom/passes/sccp/sccp.py

```diff
@@ -305,6 +311,9 @@ def _replace_constants(self, inst: IRInstruction):
                 inst.opcode = "jmp"
                 inst.operands = [target]
                 self.cfg_dirty = True
+                for bb in inst.parent.cfg_out:
+                    if bb.label == target:
+                        self.cfg_phi_fix_needed.add(bb)
```

Comment:
Why not `self._fix_phi_bb_r(bb, OrderedSet())` here and remove all the need for `self.cfg_phi_fix_needed` and iterating later?

Reply:
The reason was not to iterate over some basic block multiple times. Since solutions only detects start from which the occurrence of faulty phi node could be possible but not the end there could be a lot of overlap. In this case since I iterate over possible positions of the faulty phi nodes at the end I visit every possible basic block only once.